### PR TITLE
Improve separation between SDK and GoogleTest CMake configuration #12

### PR DESCRIPTION
**Pull request by kdowney-lot49**
Broke out all unit testing and API doc generation code into sub-directories and switched to using presets. Since it's simpler to run now, also cleaned up the CI workflow to take advantage of the new setup.

#12

### CODE REVIEW

File: docs/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+  if (DOXYGEN_FOUND)
+      # Note the option ALL which allows to build the docs together with the application
+      set(DOXYGEN_CFG ${CMAKE_CURRENT_SOURCE_DIR}/../Doxyfile)
+      add_custom_target( doc_doxygen ALL
+          COMMAND ${DOXYGEN_EXECUTABLE} ${DOXYGEN_CFG}
+          WORKING_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}
+          COMMENT "Generating API documentation with Doxygen"
+          VERBATIM )
+  else (DOXYGEN_FOUND)
+    message("Doxygen need to be installed to generate the doxygen documentation")
```
Comment (BartoszCiesla) [inline docs/CMakeLists.txt]: I think in this case it should be `FATAL_ERROR`

File: docs/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,16 @@
+# set up Doxygen for API doc generation
+if(BUILD_DOCS)
+  find_package(Doxygen)
+
+  if (DOXYGEN_FOUND)
+      # Note the option ALL which allows to build the docs together with the application
+      set(DOXYGEN_CFG ${CMAKE_CURRENT_SOURCE_DIR}/../Doxyfile)
+      add_custom_target( doc_doxygen ALL
```
Comment (BartoszCiesla) [inline docs/CMakeLists.txt]: I would use `docs` or `doxygen` target name - but it is up to you ;-)

File: .vscode/settings.json [hunk 3]
```diff
@@ -3,5 +3,6 @@
         "activityBar.background": "#571425",
         "titleBar.activeBackground": "#7A1C34",
         "titleBar.activeForeground": "#FEFBFC"
-    }
+    },
+    "testMate.cpp.test.executables": "${workspaceFolder}/build/tests/**/*_test"
 }
```
Comment (BartoszCiesla) [inline .vscode/settings.json]: Please add empty line here

File: CMakePresets.json [hunk 4]
```diff
@@ -4,9 +4,42 @@
...
+    ],
+    "buildPresets": [
+        {
+          "name": "ninja",
+          "configurePreset": "ci",
+          "displayName": "Build with Ninja",
+          "description": "Build with Ninja"
         }
     ]
 }
```
Comment (BartoszCiesla) [inline CMakePresets.json]: Please add empty line here

**Review by hubcio — APPROVED** [review]

**Pull request merged**